### PR TITLE
Product collection segments endpoint

### PR DESCRIPTION
**Pull request by wiewiurdp**

### CODE REVIEW

File: module/product/src/Persistence/Dbal/Query/DbalProductQuery.php [hunk 1]
```diff
@@ -120,6 +182,62 @@ public function findProductIdByAttributeId(AttributeId $attributeId, ?Uuid $valu
...
+     */
+    public function findProductIdsBySkus(array $skus = []): array
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->select('id')
+            ->from(self::PRODUCT_TABLE);
+
+        if ($skus) {
+            $qb->andWhere($qb->expr()->in('sku', ':skus'))
+                ->setParameter(':skus', $skus, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY);
```
Comment (wfajczyk) [inline module/product/src/Persistence/Dbal/Query/DbalProductQuery.php]: Change to `Connection::PARAM_STR_ARRAY`

File: module/product/src/Persistence/Dbal/Query/DbalProductQuery.php [hunk 2]
```diff
@@ -120,6 +182,62 @@ public function findProductIdByAttributeId(AttributeId $attributeId, ?Uuid $valu
...
+     * @return array
+     */
+    public function findProductIdsBySegments(array $segmentIds): array
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->select('product_id')
+            ->from(self::SEGMENT_PRODUCT_TABLE);
+        if ($segmentIds) {
+            $result = $qb->andWhere($qb->expr()->in('segment_id', ':segmentIds'))
+                ->setParameter(':segmentIds', $segmentIds, \Doctrine\DBAL\Connection::PARAM_STR_ARRAY)
```
Comment (wfajczyk) [inline module/product/src/Persistence/Dbal/Query/DbalProductQuery.php]: Change to `Connection::PARAM_STR_ARRAY`

File: module/product-collection/src/Application/Controller/Api/Element/ProductCollectionElementMultipleCreateAction.php [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\PropertyAccess\Exception\InvalidPropertyPathException;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(
+ *     name="ergonode_product_collection_element_multiple_create",
+ *     path="/collections/{collection}/elements",
```
[the remark this answers is not in the document]
Comment (wiewiurdp) [reply module/product-collection/src/Application/Controller/Api/Element/ProductCollectionElementMultipleCreateAction.php]: hmmm but in this endpoint I add more then one element, I have two endpoint one adding single element ({collection}/element) and this one which adds multiple elements ({collection}/elements)

File: module/product/src/Infrastructure/Validator/SkusValidValidator.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        $value = (string) $value;
+        $skus = array_map('trim', explode(',', $value));
+
+        foreach ($skus as $sku) {
+            if (!\Ergonode\Product\Domain\ValueObject\Sku::isValid($sku)) {
+                $this->context->buildViolation($constraint->invalidMessage)
+                    ->setParameter('{{ value }}', $sku)
+                    ->addViolation();
+            } else {
+                $result = $this->query->findBySku(new \Ergonode\Product\Domain\ValueObject\Sku($sku));
```
[the remark this answers is not in the document]
Comment (wiewiurdp) [reply module/product/src/Infrastructure/Validator/SkusValidValidator.php]: there is already Sku alias in this file, In other validators in the system it looks like this :)

File: module/product-collection/src/Application/Controller/Api/Element/ProductCollectionElementDeleteAction.php [hunk 5]
```diff
@@ -23,7 +23,7 @@
 /**
  * @Route(
  *     name="ergonode_product_collection_element_delete",
- *     path="/collections/{collection}/elements/{product}",
+ *     path="/collections/{collection}/element/{product}",
```
Comment (rprzedzik) [inline module/product-collection/src/Application/Controller/Api/Element/ProductCollectionElementDeleteAction.php]: Why change to element ?? in whole application use plural convention ??

**Pull request merged**